### PR TITLE
feat(release-notifications): Add release alert notification builder

### PR DESCRIPTION
**Pull request by scttcper**
Similar to the existing issues alert builder but moves the title to the message and adds a new release based title

example
image

### CODE REVIEW

File: src/sentry/integrations/slack/message_builder/notifications/issues.py [hunk 1]
```diff
@@ -45,12 +48,13 @@ def __init__(
...
-        return SlackIssuesMessageBuilder(
+        return SlackReleaseIssuesMessageBuilder(
             group=group,
             event=getattr(self.notification, "event", None),
             tags=self.context.get("tags", None),
             rules=getattr(self.notification, "rules", None),
             issue_details=True,
             notification=self.notification,
             recipient=self.recipient,
+            release_version=getattr(self.notification, "release_version", None),
```
Comment (scttcper) [inline src/sentry/integrations/slack/message_builder/notifications/issues.py:59]: this assumes we've attached a release_version to the notification object @barkbarkimashark

**Comment by ceorourke** [comment]
You might want to parse the release version since the full version is pretty long. We do that for the regular release notifications: https://github.com/getsentry/sentry/blob/master/src/sentry/notifications/notifications/activity/release.py#L142

**Comment by barkbarkimashark** [comment]
parse_release ...
https://github.com/getsentry/relay/blob/master/py/sentry_relay/processing.py#L205

interesting

**Review by barkbarkimashark — CHANGES_REQUESTED** [review]
blocking this from going out first. I wanna merge some changes to this before we push.

File: src/sentry/integrations/slack/message_builder/issues.py [hunk 2]
```diff
@@ -367,6 +370,81 @@ def build(self) -> SlackBody:
...
+        issue_title = build_attachment_title(obj)
+        title_url = get_title_link(
+            self.group, self.event, self.link_to_event, self.issue_details, self.notification
+        )
+        release = (
+            parse_release(self.last_release.version)["description"] if self.last_release else ""
+        )
+        release_url = absolute_uri(
+            f"/organizations/{self.group.organization.slug}/releases/{release}/?project={self.group.project_id}"
+        )
```
Comment (barkbarkimashark) [inline src/sentry/integrations/slack/message_builder/issues.py:433]: Hmmmm, I think to build the full `release_url` you have to provide the un-parsed release.version:
https://github.com/getsentry/sentry/blob/5d4ed401271eb28300edbeda065ef05fad58c5b4/src/sentry/notifications/notifications/activity/release.py#L151-L153

**Review by barkbarkimashark — CHANGES_REQUESTED** [review]

**Review by barkbarkimashark — APPROVED** [review]

**Pull request merged**